src/spaces/compartment.jl

```diff
@@ -0,0 +1,188 @@
+export CompartmentSpace
+
+struct CompartmentSpace{D,F} <: AbstractSpace
+  grid::GridSpace
+  update_vel!::F
+  dims::NTuple{D, Int}
+  periodic::Bool
```

Comment:
`periodic` becomes a type parameter, because it is the second type parameter of the field `grid`, since `GridSpace{D, periodic}`.